harness/propertyHelper.js

```diff
@@ -45,19 +45,6 @@ function verifyProperty(obj, name, desc, options) {
     return true;
   }
 
-  var names = Object.getOwnPropertyNames(desc);
-  for (var i = 0; i < names.length; i++) {
-    assert(
-      names[i] === "value" ||
-        names[i] === "writable" ||
-        names[i] === "enumerable" ||
-        names[i] === "configurable" ||
-        names[i] === "get" ||
-        names[i] === "set",
-      "Invalid descriptor field: " + names[i],
-    );
-  }
-
   assert(
     Object.prototype.hasOwnProperty.call(obj, name),
```

Comment:
Isn't this already covered by above use of `Object.getOwnPropertyDescriptor` without potentially double-triggering a proxy trap?
```suggestion
```

Reply:
Implementations can have different code paths for `Object.getOwnPropertyDescriptor` and `Object.prototype.hasOwnPropert`, so the `verifyProperty` function tries to ensure that all MOP operations are consistent. Proxies probably shouldn't be used with `verifyProperty` anyway.